### PR TITLE
[CBRD-24519] Add for PL/CSQL test cases 04 (11.4)

### PR DESCRIPTION
**Pull request by swi0110**
Refer to follow list
http://jira.cubrid.org/browse/CBRD-24901
- sql/_05_plcsql/_01_testspec/_02_declaration/_02_variable/_04_deeping/_02_char/cases/01_char_basic.sql
- sql/_05_plcsql/_01_testspec/_02_declaration/_02_variable/_04_deeping/_02_char/cases/02_char_limit.sql
- sql/_05_plcsql/_01_testspec/_02_declaration/_02_variable/_04_deeping/_03_varchar/cases/01_varchar_basic.sql
- sql/_05_plcsql/_01_testspec/_02_declaration/_02_variable/_04_deeping/_03_varchar/cases/02_varchar_limit.sql

http://jira.cubrid.org/browse/CBRD-24785
- sql/_05_plcsql/_01_testspec/_04_expression/_21_limit/cases/01_limit.sql

http://jira.cubrid.org/browse/CBRD-24903
- sql/_05_plcsql/_01_testspec/_05_bug_fix/cases/01_drop_user.sql

http://jira.cubrid.org/browse/CBRD-24946
- sql/_05_plcsql/_01_testspec/_05_bug_fix/cases/02_column_name.sql

http://jira.cubrid.org/browse/CBRD-24951
- (It adds the comment on the related issue on test cases)
sql/_05_plcsql/_01_testspec/_03_statement/_03_cursor_manipulation/_02_fetch/cases/03_03_02-06_normal_fetching_unopen.sql
sql/_05_plcsql/_01_testspec/_03_statement/_03_cursor_manipulation/_03_close/cases/03_03_03-03_normal_close_unopen.sql
sql/_05_plcsql/_01_testspec/_03_statement/_03_cursor_manipulation/_03_close/cases/03_03_03-03_normal_close_unopen_2.sql

http://jira.cubrid.org/browse/CBRD-24954
- sql/_05_plcsql/_01_testspec/_05_bug_fix/cases/03_call_in_execute.sql

http://jira.cubrid.org/browse/CBRD-24956
- (Add to comment for related issue on test case)
sql/_05_plcsql/_01_testspec/_03_statement/_13_loop/_05_for_static_sql/cases/03_13_05-02_error_invalid_sql.sql

http://jira.cubrid.org/browse/CBRD-24957
- (Add to comment for related issue on test case)
sql/_05_plcsql/_01_testspec/_03_statement/_13_loop/_06_for_dynamic_sql/cases/03_13_06-01_normal_basic.sql
sql/_05_plcsql/_01_testspec/_03_statement/_13_loop/_06_for_dynamic_sql/cases/03_13_06-01_normal_basic_2.sql

http://jira.cubrid.org/browse/CBRD-24958
- (Add to comment for related issue on test case)
sql/_05_plcsql/_01_testspec/_03_statement/_13_loop/_06_for_dynamic_sql/cases/03_13_06-04_error_incompatible_used_val.sql

http://jira.cubrid.org/browse/CBRD-24959
- (Add to comment for related issue on test case)
sql/_05_plcsql/_01_testspec/_04_expression/_04_function_call/_03_builtin_function/_05_datetime_function/cases/04_04_03_05_bfn_datetime_date.sql

http://jira.cubrid.org/browse/CBRD-24962
- sql/_05_plcsql/_01_testspec/_04_expression/_07_sql_rowcount/_01_basic/cases/bug_after_commit_rollback.sql

http://jira.cubrid.org/browse/CBRD-24963
- sql/_05_plcsql/_01_testspec/_03_statement/_03_cursor_manipulation/_04_open_for/cases/bug_sys_refcursor_raise.sql

### CODE REVIEW

File: sql/_05_plcsql/_01_testspec/_05_bug_fix/cases/02_column_name.sql [hunk 1]
```diff
@@ -0,0 +1,16 @@
+--+ server-message on
+-- Verified for CBRD-24946
+-- Field lookup failure when column name is of <table>.<column> form in the select list
+
+create or replace procedure test_proc as
+begin
+    for r in (select t.name from db_user t) loop
```
Comment (kwonhoil) [inline sql/_05_plcsql/_01_testspec/_05_bug_fix/cases/02_column_name.sql]: select t.name from db_user t order by t.name

Comment (swi0110) [reply sql/_05_plcsql/_01_testspec/_05_bug_fix/cases/02_column_name.sql]: Thx, I reflect that.

**Review by ssihil — APPROVED** [review]

File: sql/_05_plcsql/_01_testspec/_04_expression/_21_limit/cases/01_limit.sql [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+end;
+
+call limit_test1();
+
+
+
+create or replace procedure limit_test2 as
+	col varchar;
+begin
+	select 'limit test 2' into col from dual limit 1;
```
Comment (kwonhoil) [inline sql/_05_plcsql/_01_testspec/_04_expression/_21_limit/cases/01_limit.sql]: select 'limit test 2' into col from t  limit 1;

File: sql/_05_plcsql/_01_testspec/_04_expression/_07_sql_rowcount/_01_basic/cases/bug_after_commit_rollback.sql [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+begin
+
+	select charset_id into i from db_charset where charset_name='utf8';
+	dbms_output.put_line(sql%rowcount);
+	commit;
+	dbms_output.put_line(sql%rowcount);
+end;
+
+call t();
+
```
Comment (kwonhoil) [inline sql/_05_plcsql/_01_testspec/_04_expression/_07_sql_rowcount/_01_basic/cases/bug_after_commit_rollback.sql:56]: create table test ( col1 int  );
insert into test values(1),(2),(3),(4),(5);

create or replace procedure t as
	i int;
begin
	insert into test values(1),(2),(3),(4),(5);
	dbms_output.put_line(sql%rowcount);
	commit;
	dbms_output.put_line(sql%rowcount);
	
	update test set col1 = col1 +1 ;
	dbms_output.put_line(sql%rowcount);
	commit;
	dbms_output.put_line(sql%rowcount);
	
	delete from test where col1 < 5 ;
	dbms_output.put_line(sql%rowcount);
	commit;
	dbms_output.put_line(sql%rowcount);
	
end;

call t();


drop table test ;
create table test ( col1 int  );
insert into test values(1),(2),(3),(4),(5);

create or replace procedure t as
	i int;
begin
	insert into test values(1),(2),(3),(4),(5);
	dbms_output.put_line(sql%rowcount);
	rollback;
	dbms_output.put_line(sql%rowcount);
	
	update test set col1 = col1 +1 ;
	dbms_output.put_line(sql%rowcount);
	rollback;
	dbms_output.put_line(sql%rowcount);
	
	delete from test where col1 < 5 ;
	dbms_output.put_line(sql%rowcount);
	rollback;
	dbms_output.put_line(sql%rowcount);
	
end;

call t();

File: sql/_05_plcsql/_01_testspec/_02_declaration/_02_variable/_04_deeping/_03_varchar/cases/02_varchar_limit.sql [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	
+	dbms_output.put_line('varchar size 1');
+	dbms_output.put_line('value:   ' || vch_1);
+	dbms_output.put_line('length:  ' || vch_length);
+end;
+
+call varchar_limit3('a');
+--error, size over
+call varchar_limit3('abc');
+
```
Comment (kwonhoil) [inline sql/_05_plcsql/_01_testspec/_02_declaration/_02_variable/_04_deeping/_03_varchar/cases/02_varchar_limit.sql:61]: add hangul;

File: sql/_05_plcsql/_01_testspec/_02_declaration/_02_variable/_04_deeping/_03_varchar/cases/02_varchar_limit.sql [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+drop procedure varchar_limit2;
+drop procedure varchar_limit3;
+
```
Comment (kwonhoil) [inline sql/_05_plcsql/_01_testspec/_02_declaration/_02_variable/_04_deeping/_03_varchar/cases/02_varchar_limit.sql:72]: add hangul;

File: sql/_05_plcsql/_01_testspec/_02_declaration/_02_variable/_04_deeping/_02_char/cases/01_char_basic.sql [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+end;
+
+call char_test1('select cast(''abc'' as char(3)) from dual;');
+-- error, size over
+call char_test1('select cast(''abc'' as char(21)) from dual;');
+
+
+create or replace procedure char_test2() as
+	num_1 character(20) := 3;
+	num_2 character(5) := 8;
```
Comment (kwonhoil) [inline sql/_05_plcsql/_01_testspec/_02_declaration/_02_variable/_04_deeping/_02_char/cases/01_char_basic.sql]: num_1 character(20) := 2;
	num_2 character(5) := 10;

File: sql/_05_plcsql/_01_testspec/_02_declaration/_02_variable/_04_deeping/_03_varchar/cases/01_varchar_basic.sql [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+	dbms_output.put_line('varchar: "' || varchar_1 ||'"');
+end;
+
+call varchar_test1('select cast(''abc'' as varchar(3)) from dual;');
+
+
+create or replace procedure varchar_test2() as
+	num_1 varchar(20) := 3;
+	num_2 varchar(5) := 8;
+begin
```
Comment (kwonhoil) [inline sql/_05_plcsql/_01_testspec/_02_declaration/_02_variable/_04_deeping/_03_varchar/cases/01_varchar_basic.sql:19]: num_1 varchar(20) := 2;
	num_2 varchar(5) := 10;

File: sql/_05_plcsql/_01_testspec/_04_expression/_21_limit/cases/01_limit.sql (hunk 2 above)
Comment (swi0110) [reply sql/_05_plcsql/_01_testspec/_04_expression/_21_limit/cases/01_limit.sql]: reflected review

File: sql/_05_plcsql/_01_testspec/_02_declaration/_02_variable/_04_deeping/_02_char/cases/01_char_basic.sql (hunk 6 above)
Comment (swi0110) [reply sql/_05_plcsql/_01_testspec/_02_declaration/_02_variable/_04_deeping/_02_char/cases/01_char_basic.sql]: num_1, num_2 is char type.
this case is not auto-cast by integer.
Therefore I add a num_1 > num_2 and num_2 > num_1 on this test case.

File: sql/_05_plcsql/_01_testspec/_02_declaration/_02_variable/_04_deeping/_03_varchar/cases/01_varchar_basic.sql (hunk 7 above)
Comment (swi0110) [reply sql/_05_plcsql/_01_testspec/_02_declaration/_02_variable/_04_deeping/_03_varchar/cases/01_varchar_basic.sql:19]: num_1, num_2 is varchar type.
this case is not auto-cast by integer.
Therefore I add a num_1 > num_2 and num_2 > num_1 on this test case.

File: sql/_05_plcsql/_01_testspec/_04_expression/_07_sql_rowcount/_01_basic/cases/bug_after_commit_rollback.sql (hunk 3 above)
Comment (swi0110) [reply sql/_05_plcsql/_01_testspec/_04_expression/_07_sql_rowcount/_01_basic/cases/bug_after_commit_rollback.sql:56]: Thx, I reflected that

File: sql/_05_plcsql/_01_testspec/_02_declaration/_02_variable/_04_deeping/_03_varchar/cases/01_varchar_basic.sql (hunk 7 above)
Comment (swi0110) [reply sql/_05_plcsql/_01_testspec/_02_declaration/_02_variable/_04_deeping/_03_varchar/cases/01_varchar_basic.sql:19]: reflected review

File: sql/_05_plcsql/_01_testspec/_02_declaration/_02_variable/_04_deeping/_02_char/cases/01_char_basic.sql (hunk 6 above)
Comment (swi0110) [reply sql/_05_plcsql/_01_testspec/_02_declaration/_02_variable/_04_deeping/_02_char/cases/01_char_basic.sql]: add_plcsql_04

**Review by kwonhoil — APPROVED** [review]

**Pull request merged**